docs/contributors/index.md

```diff
@@ -0,0 +1,243 @@
+```
+
+In development mode, you will also need to remove the symlink created by
+`jupyter labextension develop` command. To find its location, you can run
+`jupyter labextension list` to figure out where the `labextensions` folder is
+located. Then you can remove the symlink named `jupyter-scheduler` within that
+folder.
+
+## Testing
+
+### Server tests
+
+This extension is using [Pytest](https://docs.pytest.org/) for Python code testing.
+
+Install test dependencies (needed only once):
+
+```sh
+pip install -e ".[test]"
+```
+
+To execute them, run:
+
+```sh
+pytest -vv -r ap --cov jupyter_scheduler
+```
+
+### Frontend tests
+
+This extension is using [Jest](https://jestjs.io/) for JavaScript code testing.
+
+To execute them, execute:
+
+```sh
+jlpm
+jlpm test
+```
+
+### Integration tests
+
+This extension uses Playwright for the integration tests (aka user level tests).
+More precisely, the JupyterLab helper
+[Galata](https://github.com/jupyterlab/jupyterlab/tree/master/galata) is used to
+handle testing the extension in JupyterLab.
+
+More information are provided within the
+[ui-tests](https://github.com/jupyter-server/jupyter-scheduler/tree/main/ui-tests)
+README.
+
+## Documentation
+
+First, ensure GNU Make is installed locally, and then install the `docs` extras:
+
+```
+pip install -e ".[docs]"
+```
+
+Documentation is built with the [Sphinx](https://www.sphinx-doc.org/en/master/)
+documentation generator, via the following command run from the project root:
+
+```
+make -C docs html
+```
+
+Documentation source files are written in
+[MyST](https://myst-parser.readthedocs.io/en/latest/index.html), a rich and more
+expressive flavor of Markdown. These files are located under `docs/`.
+
+The built documentation files are generated under `docs/_build/html` can be
+accessed in the browser via a file URI, e.g.
+`file:///Users/dlq/workplace/jupyter-scheduler/docs/_build/html/index.html`.  We
+recommend bookmarking this if you will be editing documentation frequently.
+
+Sphinx by default only rebuilds files it detects were changed, though this
+detection logic is sometimes faulty. To force a full rebuild of the
+documentation:
+
+```
+make -C docs clean && make -C docs html
+```
+
+## Releasing
+
+Releases should be done via [Jupyter Releaser](https://github.com/jupyter-server/jupyter_releaser).
+If you have admin permissions on the repository, you can go to the "Actions"
+panel on GitHub and follow the below instructions to release a new version of
+Jupyter Scheduler.
+
+1. Click the "Step 1: Prep Release" workflow on the left-hand panel, and then
+click "Run workflow". This will open a form. Replace the "Next Version
+Specifier" with the next version of Jupyter Scheduler (e.g. 1.2.3). Do not
+include the "v" prefix. "Branch to Target" should be replaced with the branch
+that should be released. Usually this will be `main`. Then click "Run workflow"
+(the green button) to submit the form and run the workflow.
```

Comment:
```suggestion
1. Select the "Step 1: Prep Release" workflow on the left-hand panel, and then
select "Run workflow". This will open the workflow form. Replace the "Next Version
Specifier" with the next version of Jupyter Scheduler (e.g. 1.2.3). "Branch to Target" 
should be replaced with the branch that should be released. Usually this will be `main`. 
Then select "Run workflow" button to run the workflow.
```